--- a/Makefile	2025-08-07 23:40:09.638319479 +0800
+++ b/Makefile	2025-08-07 23:43:44.456888889 +0800
@@ -1403,9 +1403,11 @@
   ifneq ($(call clang-ifversion, -ge, 0500, y), y)
 	@echo Cannot use CONFIG_LTO_CLANG: requires clang 5.0 or later >&2 && exit 1
   endif
-  ifneq ($(ld-name), lld)
-    ifneq ($(call gold-ifversion, -ge, 112000000, y), y)
-	@echo Cannot use CONFIG_LTO_CLANG: requires LLD or GNU gold 1.12 or later >&2 && exit 1
+  ifneq ($(ld-name), mold)
+    ifneq ($(ld-name), lld)
+      ifneq ($(call gold-ifversion, -ge, 112000000, y), y)
+        @echo Cannot use CONFIG_LTO_CLANG: requires LLD or GNU gold 1.12 or later >&2 && exit 1
+      endif
     endif
   endif
 endif
